exonum/src/runtime/mod.rs

```diff
@@ -138,24 +210,39 @@ impl<'a> Default for ArtifactInfo<'a> {
     }
 }
 
+/// An accessory structure that aggregates root objects hashes of runtime service
+/// information schemas with the root hash of runtime information schema itself.
 #[derive(Debug, PartialEq, Default)]
 pub struct StateHashAggregator {
+    /// List of hashes of the root objects of runtime information schemas.
     pub runtime: Vec<Hash>,
+    /// List of hashes of the root objects of service instances schemas.
     pub instances: Vec<(ServiceInstanceId, Vec<Hash>)>,
 }
 
+/// The one who causes the transaction execution.
 #[derive(Debug, PartialEq)]
 pub enum Caller {
-    Transaction { hash: Hash, author: PublicKey },
+    /// A usual transaction from Exonum client, authorized by his key pair.
+    Transaction {
+        /// The transaction hash that is used as the identifier.
```

Comment:
Not sure if it is already document prominently elsewhere, but can clarify that "The transaction _message_ hash"

Reply:
as its identifier